spec/requests/security_activity_spec.rb

```diff
@@ -63,6 +63,38 @@
       expect_event SecurityActivity::ADDITIONAL_FACTOR_VERIFICATION_FAILURE, { analytics: "from_confirmation_email" }
       expect_event_on_security_page SecurityActivity::ADDITIONAL_FACTOR_VERIFICATION_FAILURE
     end
+
+    it "records ADDITIONAL_FACTOR_BYPASS_USED events" do
+      allow_any_instance_of(ActionDispatch::Cookies::CookieJar).to receive(:encrypted)
+        .and_return({ DeviseSessionsController::MFA_BYPASS_COOKIE_NAME => { user.email => MfaToken.generate!(user).token } })
+
+      post new_user_session_path, params: { "user[email]" => user.email, "user[password]" => user.password }
+
+      expect_event SecurityActivity::ADDITIONAL_FACTOR_BYPASS_USED
```

Comment:
Interesting, will this show up on the securty screen?
I wonder how many of these a user might have during a session...

... there will always occur alongside another even too right?
Do you think instead of being a separate event they could be a value on anoter secuirty event?

So like:
a login event with an ADDITIONAL_FACTOR_BYPASS_USED: true parameter?

Reply:
Currently I'm only thinking of surfacing the `GENERATED` event in the security log, as eg we don't show the `ADDITIONAL_FACTOR_VERIFICATION_SUCCESS` ones